src/lib/profiles/service-directory/ServiceDirectory.cpp

```diff
@@ -358,10 +362,41 @@ WEAVE_ERROR WeaveServiceManager::init(WeaveExchangeManager *aExchangeMgr,
 
     clearCacheState();
 
+#if WEAVE_CONFIG_PERSIST_SERVICE_DIRECTORY
+
+    if (platform::PersistentServiceDirExist(kMsgType_ServiceEndpointResponse))
```

Comment:
Recommend renaming method as IsPersistentServiceDirPresent(..). Also, potentially define a local variable version and assigning to it the messageType before passing to the function.